decentraland/unity-renderer
# feat: Implement manual connection to Voice Chat

## What does this PR change?
Due the technical problems related to browser permission, some players may need to manually connect to the voice chat.
So far, the system automatically connect to it, and if the player decides to leave, the voice chat HUD dissapears.
Now, it will always be present, even if your are not connected to it.

In the case the player is not connected, the HUD will become a button to Join in a single click.

![Screenshot 2022-08-12 at 17.24.46.png](https://images.zenhubusercontent.com/5d91f7e9df37660001a11578/027e7e59-9231-468a-8e6c-cba1be10646b)

**Acceptance Criteria**
- [x] The system will always automatically try to connect to the voice chat
- [x] When voice chat is disconnected, the HUD will be a button
- [x] Disconnecting from the voice chat will turn the HUD into a button to rejoin

## How to test the changes?
1. Go to: https://play.decentraland.zone/?renderer-branch=feat/implement-manual-connection-to-voice-chat
2. Test the voice chat with other users pressing T (or keeping pressed the microphone button in the Voice Chat bar) and talking.

![image](https://user-images.githubusercontent.com/64659061/185200345-9b1423ad-509b-4830-9fba-f8c37c381a93.png)

3. If it is not able to send the audio, check that you have received a warning notification informing about that and you will see the Voice Chat system is disconnected (a "JOIN VOICE CHAT" button will be placed in the taskbar).

![image](https://user-images.githubusercontent.com/64659061/185200021-8f21efa3-643b-45b6-a329-ed36a9ee9e6a.png)

4. If you click on this "JOIN VOICE CHAT" button, you will be able to reconnect and you should see the voice chat bar actived again in the taskbar. In this point, test the voice chat with other users pressing T (or keeping pressed the microphone button in the Voice Chat bar) and talking. Notice the audio sending is working correctly now.

![image](https://user-images.githubusercontent.com/64659061/185201586-0d4e4dba-829a-47bb-8ade-bbd82ba4552c.png)

5. Click on the LEAVE button (through the taskbar or through the Voice Chat window) and the Voice Chat system should be disconnected (you will see the "JOIN VOICE CHAT" button again in the taskbar).

![image](https://user-images.githubusercontent.com/64659061/185201794-4bff16cf-9de2-40e9-8113-7dc594db7058.png)

## Our Code Review Standards
https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md
